leaderelection/leaderelection.go

```diff
@@ -47,37 +47,44 @@ type ElectionManager interface {
 }
 
 type Manager struct {
-	log        *logp.Logger
-	leader     *le.LeaderElector
-	wg         *sync.WaitGroup
-	cancelFunc context.CancelFunc
-	kubeClient k8s.Interface
+	log          *logp.Logger
+	leader       *le.LeaderElector
+	wg           *sync.WaitGroup
+	cancelFunc   context.CancelFunc
+	kubeClient   k8s.Interface
+	k8sAvailable bool
 }
 
-func NewLeaderElector(log *logp.Logger, cfg *config.Config) (ElectionManager, error) {
-	kubeClient, err := providers.KubernetesProvider{}.GetClient(cfg.KubeConfig, kubernetes.KubeClientOptions{})
+func NewLeaderElector(log *logp.Logger, cfg *config.Config) ElectionManager {
```

Comment:
Instead, I would define a new struct `DummyManager` that statically implements the same `ElectionManager` interface. In case you are not in a k8s cluster, we will use that manager instead.
That way, we avoid adding non-k8s logic to a k8s-oriented file.
Also, unit testing will get easier.

Reply:
Add tests